powerauth-java-crypto/src/test/java/com/wultra/security/powerauth/crypto/lib/util/GenerateVectorDataTest.java

```diff
@@ -85,6 +89,150 @@ public static void setUp() {
         objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
     }
 
+    /**
+     * Characters used in {@link #getRandomString(int, int, String[])} method.
+     */
+    static final int[] RANDOM_STRING_CHARS = {
+            32, 33, 35, 38, 39, 40, 41, 43, 45, 46, 47,
+            48, 49, 50, 51, 52, 53, 54, 55, 56, 57,
+            65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90,
+            97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122
+    };
+
+    /**
+     * Generate random string or select random string from predefined set.
+     * @param minLength Minimum string length.
+     * @param maxLength Maximum string length.
+     * @param predefinedSet Predefined set of strings.
+     * @return Random string.
+     */
+    String getRandomString(int minLength, int maxLength, String[] predefinedSet) {
+        boolean fixedLength = minLength == maxLength;
+        int length = fixedLength ? maxLength : maxLength - minLength;
+        length = RANDOM.nextInt(length);
+        if (length == 0 && predefinedSet != null) {
+            return predefinedSet[RANDOM.nextInt(predefinedSet.length)];
+        }
+        if (!fixedLength) {
+            length += minLength;
+        }
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < length; i++) {
```

Comment:
If I use `minLength=5` and `maxLength=5`, the maximum random number for `lenght` is 4. Then we append 4 codepoints. I think you wanted to append 5 codepoints in this case, right?

Reply:
You're right. I fixed this and also increased probability of selection from the predefined set.